### PR TITLE
Feat/flash message

### PR DESCRIPTION
**Pull request by cmokbel1**
flash message only flashes success messages or error that get put out from the server. Anything user related will be flashed beneath their respective inputs with red text

### CODE REVIEW

File: frontend/src/components/FlashMessage.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+
+export const FlashMessage = ({ messageState, returnError }) => {
+    if (messageState) {
+        return (
+            <div className="container mb-4" style={{ backgroundColor: 'green', color: 'snow' }}>
```
Comment (1gm) [inline frontend/src/components/FlashMessage.js:6]: I think you should be using the alert alert-success and alert alert-danger classes for success/failure, but also I feel like the FlashMessage should only apply to successes. E.g. Successfully logged in, successfully added item, successfully deleted list...

Errors should still be local to the area where they occured, e.g. input required errors

File: frontend/src/components/ListDetail.js [hunk 2]
```diff
@@ -2,27 +2,27 @@ import { useEffect, useState } from 'react';
...
 import { Item } from './Item';
 
-export const ListDetail = ({ id, name, completed, items, handleUpdate, removeList }) => {
-    const [messageState, setMessageState] = useState('');
-    const [errorMessageState, setErrorMessageState] = useState('');
+export const ListDetail = ({ id, name, completed, items, handleUpdate, removeList, setReturnError, setMessageState }) => {
     const [itemsState, setItemsState] = useState(items ? items : []);
     const [newItemName, setNewItemName] = useState('');
     const [currentName, setCurrentName] = useState(name)
+    const [errorMessage, setErrorMessage] = useState(null)
```
Comment (1gm) [inline frontend/src/components/ListDetail.js:9]: Why is the default state for errorMessage, setErrorMessage a null but for the other string (newitemName) it's an empty string?

File: frontend/src/components/ToDoLists.js [hunk 3]
```diff
@@ -2,13 +2,12 @@ import { getList, getLists, addList, updateListName, deleteList } from '../http/
 import { useState, useEffect } from 'react';
 import { ListDetail } from './ListDetail';
 
-export const ToDoLists = ({ userState }) => {
+export const ToDoLists = ({ userState, setReturnError, setMessageState }) => {
     const [lists, setLists] = useState([]);
     const [selectedList, setSelectedList] = useState();
-
+    const [errorMessage, setErrorMessage] = useState('')
```
Comment (1gm) [inline frontend/src/components/ToDoLists.js:8]: This is inconsistent with how the state is handled in ListDetail

File: frontend/src/components/login.js [hunk 4]
```diff
@@ -15,7 +16,8 @@ const Login = ({ setUserState }) => {
             if (res.name) {
                 setUserState(res.name);
             } else {
-                setErrorState(res);
+
```
Comment (1gm) [inline frontend/src/components/login.js:19]: Why the additional new line?

File: frontend/src/components/login.js [hunk 5]
```diff
@@ -1,10 +1,11 @@
 import { useState } from 'react';
 import { loginUser } from '../http/user'
 
-const Login = ({ setUserState }) => {
+const Login = ({ setUserState, setReturnError }) => {
```
Comment (1gm) [inline frontend/src/components/login.js:4]: Where is setReturnError being used?

File: frontend/src/components/FlashMessage.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+
+
+export const FlashMessage = ({ messageState, returnError }) => {
```
Comment (1gm) [inline frontend/src/components/FlashMessage.js:3]: I don't understand how this component flashes a message? It doesn't seem to be able to flash?

File: frontend/src/components/ListDetail.js (hunk 2 above)
Comment (cmokbel1) [reply frontend/src/components/ListDetail.js:9]: addressed

File: frontend/src/components/login.js (hunk 5 above)
Comment (cmokbel1) [reply frontend/src/components/login.js:4]: removed

File: frontend/src/components/FlashMessage.js (hunk 6 above)
Comment (cmokbel1) [reply frontend/src/components/FlashMessage.js:3]: it is timed out

**Comment by 1gm** [comment]
Looks mostly good to me - I don't agree with the FlashMessage implementation but it's not worth debating.

**Pull request merged**